gadgets/trace_dns/program.bpf.c

```diff
@@ -328,6 +329,8 @@ int ig_trace_dns(struct __sk_buff *skb)
 
 	// Enrich event with process metadata
 	struct sockets_value *skb_val = gadget_socket_lookup(skb);
+	if (gadget_should_discard_data_by_skb(skb_val))
+		return 0;
```

Comment:
A DNS query could be accepted and the DNS answer could be discarded if the UDP socket was closed before receiving the answer and re-created with a different pid or different comm.

Then, the entries in query_map might not get cleaned up.

But this problem existed before your PR and the image-based dns gadget should do the garbage collection in the WASM module, like it was done for the builtin gadget in pkg/gadgets/trace/dns/tracer/gc.go.

Reply:
I see, so there is not any action on this PR, right? I'll create an issue about that after this PR is done.

Reply:
Correct. I just wrote it down not to forget.

(I have not finished reviewing the PR)